docs/src/adoc/joins.adoc

```diff
@@ -0,0 +1,39 @@
+= Join Mapping
+
+Joining multiple tables together is a very common database
+task.  It is also where the mismatch between the relational
+model and Java's object model starts to rear its ugly head.
+
+Here we present a couple of strategies for retrieving
+results from more complicated rows.
+
+== JoinRowMapper
```

Comment:
This would be a good place to describe folds as well. Maybe show the new sexy way first, and then show them how to "drop down" to folds if join mapping doesn't suffice.